x/nodes/keeper/valStateChanges_test.go

```diff
@@ -350,6 +352,129 @@ func TestValidatorStateChange_EditAndValidateStakeValidatorAfterNonCustodialUpgr
 	}
 }
 
+func runStake(ctx sdk.Context,
+	k Keeper, msg types.MsgStake, signer crypto.PublicKey) sdk.Error {
+	validator := types.NewValidator(
+		sdk.Address(msg.PublicKey.Address()),
+		msg.PublicKey,
+		msg.Chains,
+		msg.ServiceUrl,
+		sdk.ZeroInt(),
+		msg.Output)
+	err := k.ValidateValidatorStaking(
+		ctx, validator, msg.Value, sdk.Address(signer.Address()))
+	if err != nil {
+		return err
+	}
+	return k.StakeValidator(ctx, validator, msg.Value, signer)
+}
+
+func Test_OutputAddressChange(t *testing.T) {
+	ctx, _, k := createTestInput(t, true)
+
+	// Enable EditStake
+	codec.UpgradeHeight = -1
+
+	stakeAmount := sdk.NewCoin(k.StakeDenom(ctx), sdk.NewInt(k.MinimumStake(ctx)))
+	outputAccount := getRandomPubKey()
+	outputAddress := sdk.Address(outputAccount.Address())
+	msgTemplate := types.MsgStake{
+		Chains:     []string{"0021", "0040"},
+		ServiceUrl: "https://www.pokt.network:443",
+		Value:      stakeAmount.Amount,
+	}
+
+	// Create and fund accounts
+	operatorAcc1 := getRandomPubKey()
+	operatorAcc2 := getRandomPubKey()
+	operatorAcc3 := getRandomPubKey()
+	operatorAddr1 := sdk.Address(operatorAcc1.Address())
+	operatorAddr2 := sdk.Address(operatorAcc2.Address())
+	operatorAddr3 := sdk.Address(operatorAcc3.Address())
+	assert.Nil(t, fundAccount(ctx, k, operatorAddr1, stakeAmount))
+	assert.Nil(t, fundAccount(ctx, k, operatorAddr2, stakeAmount))
+	assert.Nil(t, fundAccount(ctx, k, outputAddress, stakeAmount))
+
+	// Stake two nodes before NCUST
+	msgTemplate.PublicKey = operatorAcc1
```

Comment:
I think you should copy `msgStake = proto.Clone(msgStakeTemplate)` and then update it.

It doesn't matter from a test passing perspective, but just makes more sense and less prone to error.

Reply:
Ditto below

Reply:
Yeah, if it's named `msgTemplate`, it should behave as a template.  I found a better way to call `runStake`.